core/monitor/auction.go

```diff
@@ -110,6 +111,26 @@ func (a *AuctionState) StartPriceAuction(t time.Time, d *types.AuctionDuration)
 	a.stateChanged = true
 }
 
+func (a *AuctionState) StartGovernanceSuspensionAuction(t time.Time) {
+	a.mode = types.MarketTradingModeSuspendedViaGovernance
+	a.trigger = types.AuctionTriggerGovernanceSuspension
+	a.start = true
+	a.stop = false
+	a.begin = &t
+	a.end = &types.AuctionDuration{Duration: math.MaxInt64}
```

Comment:
I don't think we need to set this duration to `MaxInt64`, you can set it to `0` and it gets treated as forever. Eg in `ExpiresAt()`:
```
if a.end == nil || a.end.Duration == 0 { // not time limited
	return nil
}
```